### PR TITLE
Making covscan happy

### PR DESCRIPTION
**Pull request by marusak**

### CODE REVIEW

File: src/common/cockpitpipe.c [hunk 1]
```diff
@@ -1234,8 +1234,10 @@ spawn_setup (gpointer data)
   prctl (PR_SET_PDEATHSIG, SIGHUP);
 #endif
 
-  if (flags & COCKPIT_PIPE_STDERR_TO_STDOUT)
-    dup2 (1, 2);
+  if (flags & COCKPIT_PIPE_STDERR_TO_STDOUT) {
+    int r = dup2 (1, 2);
+    g_assert (r == 2); /* that should really never fail */
```
Comment (allisonkarlitskaya) [inline src/common/cockpitpipe.c:1239]: `g_assert_cmpint()` would be mildly nicer here

Comment (allisonkarlitskaya) [inline src/common/cockpitpipe.c:1239]:
```suggestion
    g_assert_cmpint (r, ==, 2); /* that should really never fail */
```

Comment (marusak) [reply src/common/cockpitpipe.c:1239]: If I recall correctly, covscan cannot recognize `g_assert_cmpint`, but surely I can change it to that if it is better and then waive it in covscan.

Comment (martinpitt) [reply src/common/cockpitpipe.c:1239]: Correct, I just waived a false coverity positive yesterday that completely ignored `g_assert_cmpint()`.

Comment (marusak) [reply src/common/cockpitpipe.c:1239]: So what do you think, which would be better? @allisonkarlitskaya @martinpitt

Comment (allisonkarlitskaya) [reply src/common/cockpitpipe.c:1239]: I don't really care.  `g_assert_cmpint()` is definitely better, but if we put this here only to make Coverity happy, then perhaps we should actually put something that makes Coverity happy :shrug:

Comment (martinpitt) [reply src/common/cockpitpipe.c:1239]: Exactly. At the moment I'm not even sure that coverity properly recognizes `g_assert()`. But _cmpint doesn't help here.

**Review by martinpitt — APPROVED** [review]

**Pull request merged**